src/platform/utilities/tests/api/index.unit.spec.js

```diff
@@ -28,9 +28,150 @@ describe('test wrapper', () => {
       server.resetHandlers();
       expected = undefined;
+      sessionStorage.removeItem('shouldRedirectExpiredSession');
+    });
+
+    it('should behave as if in production', async () => {
+      await apiRequest('/status', {}, null, null, mockEnv);
+      expect(mockEnv.isProduction.called).to.be.true;
+    });
+
+    it('should redirect to /session-expired if in production and session expired (401)', async () => {
+      server.use(
+        rest.get('*', (req, res, ctx) =>
+          res(
+            ctx.status(401),
+            ctx.json({ errors: [{ status: '401', title: 'Unauthorized' }] }),
+          ),
+        ),
+      );
+
+      sessionStorage.setItem('shouldRedirectExpiredSession', 'true');
+
+      Object.defineProperty(window, 'location', {
+        value: {
+          pathname: '/some-other-page',
+          assign: sinon.stub(),
+        },
+        writable: true,
+      });
+
+      try {
+        await apiRequest(
+          '/status',
+          { headers: { 'Content-Type': 'application/json' } },
+          null,
+          null,
+          mockEnv,
+        );
+      } catch (error) {
+        expect(mockEnv.isProduction.called).to.be.true;
+        expect(window.location).to.eql('/session-expired');
+      }
+    });
+
+    it('should NOT redirect if not in production, even if session expired (401)', async () => {
+      const nonProdEnv = {
+        ...environment,
+        isProduction: sinon.stub().returns(false),
+      };
+
+      server.use(
+        rest.get('*', (req, res, ctx) =>
+          res(
+            ctx.status(401),
+            ctx.json({ errors: [{ status: '401', title: 'Unauthorized' }] }),
+          ),
+        ),
+      );
+
+      sessionStorage.setItem('shouldRedirectExpiredSession', 'true');
+
+      Object.defineProperty(window, 'location', {
+        value: {
+          pathname: '/some-other-page',
+          assign: sinon.stub(),
+        },
+        writable: true,
+      });
+
+      try {
+        await apiRequest(
+          '/status',
+          { headers: { 'Content-Type': 'application/json' } },
+          null,
+          null,
+          nonProdEnv,
+        );
+      } catch (error) {
+        // Verify there is no redirect outside of production
+        expect(nonProdEnv.isProduction.called).to.be.true;
+        expect(window.location.assign.called).to.be.false;
+      }
+    });
+
+    it('should not redirect to /session-expired if on /declined page (status: 401)', async () => {
+      server.use(
+        rest.get('*', (req, res, ctx) =>
+          res(
+            ctx.status(401),
+            ctx.json({ errors: [{ status: '401', title: 'Unauthorized' }] }),
+          ),
+        ),
+      );
+
+      sessionStorage.setItem('shouldRedirectExpiredSession', 'true');
+
+      Object.defineProperty(window, 'location', {
+        value: {
+          pathname: '/terms-of-use/declined',
+          assign: sinon.stub(),
+        },
+        writable: true,
+      });
```

Comment:
I'm not a fan of this. It should be fine for now but we should really think about alternatives.